### PR TITLE
Child care age group

### LINKED ISSUE #202: Create variable for childcare age group in NY
Per https://view.officeapps.live.com/op/view.aspx?src=https%3A%2F%2Focfs.ny.gov%2Fmain%2Fpolicies%2Fexternal%2Focfs_2019%2FLCM%2F19-OCFS-LCM-23.docx&wdOrigin=BROWSELINK

* Infant: <1.5 (center-based) or under 2 years (home-based)
* Toddler: 1.5-2 (center-based) or 2 (home-based)
* Preschooler: 3-5
* School aged: 6-12

### PR DESCRIPTION
**Pull request by tolaouk**
fixes #202

### CODE REVIEW

File: openfisca_us/variables/demographic/person.py
Comment (nikhilwoodruff) [inline openfisca_us/variables/demographic/person.py:58]: @MaxGhenis @nmrodelo feel free to override or if we should make this a general pattern, but I wonder if for all enums we should have an "UNSPECIFIED"/"UNKNOWN" value?

File: openfisca_us/variables/demographic/person.py
Comment (nikhilwoodruff) [inline openfisca_us/variables/demographic/person.py:82]: Could you add the reference link? I know it's a word online document - is there an official hosted page somewhere we can link to?

File: openfisca_us/variables/demographic/person.py
Comment (nikhilwoodruff) [inline openfisca_us/variables/demographic/person.py:88]: Ideally we'd want to compare Enums instead of comparing strings (it's slightly more efficient). For example:
```python
ccdf_care_location = person("ccdf_care_location", period)
care_locations = ccdf_care_location.possible_values # links to the Enum class
...
return select(
    [
        ((ccdf_age < 1.5) & (ccdf_care_location == care_locations.CENTER_BASED))
...
```

File: openfisca_us/variables/demographic/person.py
Comment (nikhilwoodruff) [inline openfisca_us/variables/demographic/person.py:91]: Could you use `&` and `|` instead of `and` and `or`? I think the vectorisation requires it.

**Review by nikhilwoodruff — CHANGES_REQUESTED** [review]
Thanks @tolaouk! Just left a few comments.

File: openfisca_us/variables/demographic/person.py
Comment (tolaouk) [reply openfisca_us/variables/demographic/person.py:82]: found pdf link for it :)
https://ocfs.ny.gov/main/policies/external/ocfs_2019/LCM/19-OCFS-LCM-23.pdf

File: openfisca_us/variables/demographic/person.py
Comment (MaxGhenis) [reply openfisca_us/variables/demographic/person.py:58]: What logic would we use in that case though?

File: openfisca_us/variables/demographic/person.py
Comment (MaxGhenis) [inline openfisca_us/variables/demographic/person.py:66]: Is this a function of `provider_type`?

Comment (tolaouk) [reply openfisca_us/variables/demographic/person.py:66]: Yeah actually they are
image

Comment (tolaouk) [reply openfisca_us/variables/demographic/person.py:66]: DCC_SACC -> center-based
FDC_GFDC -> home-based
LE_STD -> home-based
LE_ENH -> home-based

Comment (tolaouk) [reply openfisca_us/variables/demographic/person.py:66]: I'll need to change the formula then.

Comment (tolaouk) [reply openfisca_us/variables/demographic/person.py:66]: My current branch, child_care_age_group, doesn't have the enums for provider_type. How should I approach with this?

File: openfisca_us/variables/demographic/person.py
Comment (nikhilwoodruff) [reply openfisca_us/variables/demographic/person.py:58]: True, maybe it's better not to- it'd need an UNKNOWN path all the way through the computation tree.

File: openfisca_us/variables/demographic/person.py
Comment (MaxGhenis) [inline openfisca_us/variables/demographic/person.py:20]: @nikhilwoodruff fyi, since the rules are based on age more fine than int, e.g. 1.5 years

File: openfisca_us/variables/demographic/person.py
Comment (MaxGhenis) [inline openfisca_us/variables/demographic/person.py:62]: should we spell these out?

**Review by MaxGhenis — COMMENTED** [review]
@nikhilwoodruff any tips on diagnosing the test failures?

File: openfisca_us/variables/demographic/person.py
Comment (nikhilwoodruff) [inline openfisca_us/variables/demographic/person.py]:
```suggestion
        home_based = person("is_ccdf_home_based", period)
```

This just needs to only have the variable name and period arguments (parameters get passed automatically by OpenFisca).

File: openfisca_us/variables/demographic/person.py
Comment (nikhilwoodruff) [inline openfisca_us/variables/demographic/person.py]:
```suggestion
                ((age < 1.5) & ~home_based) | (age < 2 & home_based),
```

I think the `&` operator has precedence over `<`, so this was evaluated as `age < (1.5 & (~home_based))` (which breaks).

File: openfisca_us/variables/demographic/person.py
Comment (nikhilwoodruff) [inline openfisca_us/variables/demographic/person.py]: Same here as above

**Review by nikhilwoodruff — COMMENTED** [review]
Thanks both- just had a look at the test fails (issues were one operator precedence bug and one argument-related bug).

File: openfisca_us/variables/demographic/person.py
Comment (MaxGhenis) [inline openfisca_us/variables/demographic/person.py:71]: @nikhilwoodruff should we have default values for variables that descend from other variables with their own default values? e.g. this is based on age (default 0) and home-based (default False), so it would default to infant anyway.

**Review by MaxGhenis — APPROVED** [review]
Good to merge after Nikhil clarifies this question from me. Thanks!

File: openfisca_us/variables/demographic/person.py
Comment (MaxGhenis) [reply openfisca_us/variables/demographic/person.py]: answer: yes

**Comment by MaxGhenis** [comment]
@nikhilwoodruff I think you'll also need to update your approval

File: openfisca_us/variables/demographic/person.py
Comment (MaxGhenis) [reply openfisca_us/variables/demographic/person.py:71]: Let's actually deal with this question separately. Merging.

**Review by nikhilwoodruff — APPROVED** [review]

**Pull request merged**
